### PR TITLE
Fixed DPS310 temperature error

### PR DESCRIPTION
**Pull request by tridge**
This fixes an errata for the DPS310 temperature handling. We don't know if this applies to the DPS280 as well, so for now assume it doesn't and use hwdef.dat to select which sensor is used

### CODE REVIEW

File: libraries/AP_Baro/AP_Baro_DPS280.cpp [hunk 1]
```diff
@@ -127,6 +133,17 @@ void AP_Baro_DPS280::set_config_registers(void)
     dev->write_register(DPS280_REG_PCONF, 0x54, true); // 32 Hz, 16x oversample
     dev->write_register(DPS280_REG_TCONF, 0x54 | calibration.temp_source, true); // 32 Hz, 16x oversample
     dev->write_register(DPS280_REG_MCONF, 0x07); // continuous temp and pressure.
+
+    if (is_dps310) {
+        // work around broken temperature handling on some sensors
+        // using undocumented register writes
+        // see https://github.com/infineon/DPS310-Pressure-Sensor/blob/dps310/src/DpsClass.cpp#L442
```
Comment (WickedShell) [inline libraries/AP_Baro/AP_Baro_DPS280.cpp:140]:
```suggestion
        // see https://github.com/Infineon/DPS310-Pressure-Sensor/blob/3edb0e58dfd7691491ae8d7f6a86277b001ad93f/src/DpsClass.cpp#L442
```

Better off tagging against a commit, rather then just the branch, as these kinds of things have a tendency to get accidentally shifted as someone pushes to it.

**Comment by anbello** [comment]
I can test this on a Copter with MatekH743 (with DPS310).

What do I have to see to know if it get better with this PR?

File: libraries/AP_Baro/AP_Baro_DPS280.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_Baro/AP_Baro_DPS280.cpp:140]: if they haven't tagged then the git hash can disappear as they update. The link will still likely take them to the right file.
It is hard to predict which will be more robust

**Comment by tridge** [comment]
rebased.
@anbello unfortunately we don't know under what conditions the temperature comes out incorrectly without this fix. I'm inclined to apply the fix as it is recommended by the vendor, but if you want to test then what we're looking for is fixing the temperature reading in the baro log

**Comment by tridge** [comment]
turns out the PR didn't actually set is_dps310 due to the way probe() works. Fixed

**Comment by anbello** [comment]
@tridge I read temperature from baro that is coherent with that from IMU (3 deg max difference) without this fix. So I think shouldn't makes sense to test it, what do you think?

**Comment by peterbarker** [comment]
@anbello perhaps just test this anyway - tridge said he was inclined to merge it as the fix is recommended by the manufacturer.

**Comment by anbello** [comment]
OK, I tested the PR, I saw the same behavior of temp from baro as without the fix, coherent with that from IMU as I said above.
Maybe the fix is necessary only for some series of DPS310, but when it is not necessary it doesn't hurt. So I think it is OK to merge.

**Review by peterbarker — APPROVED** [review]

**Pull request merged**